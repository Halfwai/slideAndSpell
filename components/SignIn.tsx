import { Animated, TouchableOpacity, StyleSheet, TextInput, Alert, View } from "react-native"
import MyAppText from "@/components/MyAppText"
import { useEffect, useRef, useState } from "react"

import { supabase } from '@/lib/supabase'

import AuthButton from "@/components/AuthButton"

import { COLOURS } from '@/constants/colours'

interface SignInProps {
    setMenu: Function,
}

export default function SignIn(props: SignInProps) {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [loading, setLoading] = useState(false)

    async function signInWithEmail() {
        if(email === ''){ 
            Alert.alert('Please enter an email'); 
            return; 
        };
        if(password === ''){
            Alert.alert('Please enter a password');
            return;
        }
        setLoading(true)
        const { error } = await supabase.auth.signInWithPassword({
            email: email,
            password: password,
        })

        if (error) Alert.alert(error.message)
        setLoading(false)
    }

    return (
        <View style={styles.container}>            
            <MyAppText style={styles.title}>Sign In</MyAppText>

            <View style={styles.inputContainer}>
                <TextInput
                    placeholder="Email"
                    style={styles.input}
                    value={email}
                    onChangeText={setEmail}
                    autoComplete="email"
                    autoCorrect={false}
                    inputMode="email"
                /> 
                <TextInput
                    placeholder="Password"
                    style={styles.input}
                    value={password}
                    onChangeText={setPassword}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
            </View>

            <AuthButton text="Sign In" onPress={signInWithEmail} style={{ backgroundColor: COLOURS.green }} /> 
            <AuthButton text="Back" onPress={() => { props.setMenu("welcome") }} style={{ backgroundColor: "white", borderColor: COLOURS.green }} /> 
        </View>
    )
}

const styles =  StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
    },
    inputContainer: {
        marginBottom: 20,
        width: '100%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
    },
    input: {
        width: '90%',
        margin: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: 20,
        height: 50,
    }
})