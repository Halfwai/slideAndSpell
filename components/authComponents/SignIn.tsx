import React, { useState } from "react"
import { StyleSheet, TextInput, View } from "react-native"

// Import COLOURS
import { COLOURS } from '@/constants/colours'

// Import functions
import { signInWithEmail } from '@/utils/supabaseFunctions'

// Import components
import AuthButton from "@/components/buttons/AuthButton"
import MyAppText from "@/components/common/MyAppText"

// Setup props
interface SignInProps {
    setMenu: Function,
}

export default function SignIn(props: SignInProps) {
    // Setup sign in states
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [loading, setLoading] = useState(false)

    return (
        <View style={styles.container}>
            <MyAppText style={styles.title}>Sign In</MyAppText>

            <View style={styles.inputContainer}>
                <TextInput
                    placeholder="Email"
                    style={styles.input}
                    value={email}
                    onChangeText={setEmail}
                    autoComplete="email"
                    autoCorrect={false}
                    inputMode="email"
                />
                <TextInput
                    placeholder="Password"
                    style={styles.input}
                    value={password}
                    onChangeText={setPassword}
                    autoCorrect={false}
                    secureTextEntry={true}
                />
            </View>
            <AuthButton
                text="Sign In"
                onPress={async() => {
                    setLoading(true);
                    signInWithEmail(email, password);
                    setLoading(false);
                }}
                style={{ backgroundColor: COLOURS.green }}
            />
            <AuthButton 
                text="Back" 
                onPress={() => { 
                    props.setMenu("welcome") }
                } 
                style={{ backgroundColor: "white", borderColor: COLOURS.green }} 
                disabled={loading} 
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
    },
    inputContainer: {
        marginBottom: 20,
        width: '100%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
    },
    input: {
        width: '90%',
        margin: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: 20,
        height: 50,
    }
})