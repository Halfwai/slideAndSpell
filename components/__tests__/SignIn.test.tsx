import { render } from '@testing-library/react-native';
import SignIn from '@/components/authComponents/SignIn';

it(`SignIn Component renders correctly`, () => {
    const { toJSON } = render(
        <SignIn
            setMenu={() => {}}
        />
    );    
    expect(toJSON()).toMatchSnapshot();
});