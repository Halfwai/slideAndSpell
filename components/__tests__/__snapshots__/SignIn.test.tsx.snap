// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Quicksand",
        },
        {
          "fontSize": 53,
          "fontWeight": "bold",
          "marginBottom": 20,
        },
      ]
    }
  >
    Sign In
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="black"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 33,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="black"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 33,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#8BF0BC",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 12,
        "width": "90%",
      }
    }
    testID="authButton"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 40,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sign In
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#8BF0BC",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 12,
        "width": "90%",
      }
    }
    testID="authButton"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 40,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Back
    </Text>
  </View>
</View>
`;
